client/sftp.js

```diff
@@ -0,0 +1,83 @@
+import sftp from 'ssh2-sftp-client';
+
+export class SFTPClient {
+  constructor(timeout, logger, encoding) {
+    this.encoding = encoding;
+    this.timeout = timeout;
+    this.log = logger;
+
+    this.readOptions = {
+      readStreamOptions: {
+        flags: 'r',
+        encoding: this.encoding,
+      },
+      pipeOptions: {
+        end: false
+      }
+    };
+
+    this.client = new sftp();
+
+    this.connectionClosed = true;
+
+    this.client.on('close', this.closedConnection);
+    this.client.on('end', this.closedConnection);
+  }
+
+  async size(path) {
+    if (this.connectionClosed) throw Exception('Connect first in order to use this function');
+
+    this.log(`Getting size for file ${path}`);
+
+    return this.connection
+      .then(() => {
+        return this.client.stat(path);
+      })
+      .then((data) => {
+        this.log(`File size ${data.size}`);
+        return data.size;
+      })
+      .catch((err) => {
+        console.error(err.message);
+      });
```

Comment:
Why is `connection` a promise here? I think we should wait for that to resolve in the `access` method rather than waiting for it to resolve later.

As a side note, could we remove `.then/.catch` from here in favour of `async/await` - I'm not a fan of mixing the two.

Reply:
why I decided not to wait, is because I didn't want to tied the pattern. But maybe it makes sense for all the wrappers to just be synchronous and not use await in the caller.

https://github.com/Thomas-Smyth/ftp-tail/blob/3be20cfbe95707360236e585b0001e3d8c4181fb/index.js#L156